constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/LinearEquation.java

```diff
@@ -0,0 +1,174 @@
+package gov.nasa.jpl.aerie.constraints.model;
+
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.Segment;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import static gov.nasa.jpl.aerie.constraints.time.Interval.Inclusivity.Exclusive;
+import static gov.nasa.jpl.aerie.constraints.time.Interval.Inclusivity.Inclusive;
+
+/**
+ * A linear equation in point-slope form.
+ */
+public final class LinearEquation {
+  public final Duration initialTime;
+  public final double initialValue;
+  public final double rate;
+
+  private static final int RATE_CONVERSION_TO_MICROSECONDS = 1000000;
+
+  public LinearEquation(final Duration initialTime, final double initialValue, final double rate) {
+    this.initialTime = initialTime;
+    this.initialValue = initialValue;
+    this.rate = rate;
+  }
+
+  public double valueAt(final Duration time) {
+    return this.initialValue + this.rate*(time.minus(this.initialTime)).ratioOver(Duration.SECOND);
+  }
+
+  public LinearEquation shiftInitialTime(final Duration newInitialTime) {
+    return new LinearEquation(
+        newInitialTime,
+        this.initialValue + newInitialTime.minus(this.initialTime).ratioOver(Duration.SECOND)*this.rate,
+        this.rate
+    );
+  }
+
+  public boolean changing() {
+    return this.rate != 0;
+  }
+
+  private Optional<Duration> intersectionPointWith(final LinearEquation other) {
+    if (this.rate == other.rate) return Optional.empty();
+    return Optional.of(
+        this.initialTime.plus(
+            Duration.roundNearest(
+                RATE_CONVERSION_TO_MICROSECONDS * (other.valueAt(this.initialTime) - this.initialValue)
+                  / (this.rate - other.rate),
+                Duration.MICROSECONDS
+            )
+        )
```

Comment:
I _think_ this is equivalent, and a bit clearer:

```suggestion
        this.initialTime.plus(
            Duration.roundNearest(
                (other.valueAt(this.initialTime) - this.initialValue)
                  / (this.rate - other.rate),
                Duration.SECONDS
            )
        )
```

After reading `roundNearest`, I interpret it to mean essentially `Duration.of(float, unit)`, with the understanding that it will be rounded to the nearest microsecond.

Reply:
Ah that's a misleading function name. Ok, will do